fleetspeak/src/server/stats.go

```diff
@@ -59,6 +62,132 @@ func (s noopStatsCollector) ResourceUsageDataReceived(cd *db.ClientData, rud mpb
 func (s noopStatsCollector) KillNotificationReceived(cd *db.ClientData, kn mpb.KillNotification) {
 }
 
+
+var (
+	// Metric collectors for PrometheusStatsCollector struct
+	messagesIngested = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_messages_ingested_total",
```

Comment:
+1 to Ben's opinion in his [review](https://github.com/google/fleetspeak/pull/247#discussion_r434807335)

Fleetspeak has a notion of "service". I.e. GRR is a service in fleetspeak terms. A single Fleetspeak deployment may act as a communication middleware for a number of services. Every ingested message has a source/destination attributes, indicating the service responsible for the message (https://github.com/google/fleetspeak/blob/master/fleetspeak/src/common/proto/fleetspeak/common.proto#L9). (@bgalehouse, btw - do I understand it correctly that the service in source/destination is always the same?) We should include the service into the metric, so that later in Grafana we can filter by it.

Reply:
See [my reply](https://github.com/google/fleetspeak/pull/247#discussion_r435077477) to Ben's review.
Just to clarify - when you say that a single Fleetspeak deployment may act as a communication middleware for a number of services, do you mean concurrently? As in, multiple services (GRR server/s and multiple others) could utilize the same Fleetspeak deployment?

Reply:
Yes, multiple services can utilize the same fleetspeak deployment at the same time. Each service is identified in fleetspeak configuration by its name and fleetspeak uses the configuration to route the messages to proper client/server plugins.

Reply:
One is the service name on the client, one is the service name on the server. Which is source and which is dest depends in the obvious way on which direction the message is going.

In the case of GRR, both the client and server services are called "GRR". However, FS does not require that they be the same, and in other applications we've found making them the same to be confusing. The natural name for the daemon on the client might not be the same as the natural name for the message's next stop in the cloud.

So we should not assume that they are the same in FS. For messages received from the server from a client, the destination service name is a server service name, and this is a good parameter to break down statistics about messages arriving from clients for the server.